addons/website_sale_loyalty/controllers/main.py

```diff
@@ -90,8 +93,14 @@ def claim_reward(self, reward, **post):
             if reward_id in rewards:
                 coupon_id = coupon
         redirect = post.get('r', '/shop/cart')
-        if not coupon_id or not reward_id.exists() or reward_id.multi_product:
+        if not coupon_id or not reward_id.exists():
             return request.redirect(redirect)
+        if reward_id.multi_product and "product" in post:
+            product = request.env["product.product"].sudo().browse(int(post["product"]))
+        else:
+            product = None
+
+        request.env.context = dict(request.env.context, product=product)
```

Comment:
```suggestion
        request.update_context(...)
```

Also, please only update the context if there is effectively a product given, try to avoid context updates if not necessary.